zebra-chain/src/primitives/address.rs

```diff
@@ -119,4 +119,28 @@ impl Address {
     pub fn is_transparent(&self) -> bool {
         matches!(self, Self::Transparent(_))
     }
+
+    /// Returns the payment address for transparent or sapling addresses.
+    #[allow(clippy::unwrap_in_result)]
+    pub fn payment_address(&self) -> Option<String> {
+        use zcash_address::{ToAddress, ZcashAddress};
+
+        match &self {
+            Self::Transparent(address) => Some(address.to_string()),
+            Self::Sapling { address, network } => {
+                let data = address.to_bytes();
+                let address =
+                    ZcashAddress::from_sapling(Self::zcash_address_network(*network), data);
+                Some(address.encode())
+            }
+            Self::Unified { .. } => None,
+        }
+    }
+    /// Convert a `zebra_chain::parameters::network::Network` type into a `zcash_address::Network` type.
+    fn zcash_address_network(network: Network) -> zcash_address::Network {
+        match network {
+            Network::Mainnet => zcash_address::Network::Main,
+            Network::Testnet => zcash_address::Network::Test,
+        }
+    }
```

Comment:
`zcash_address_network()` is already implemented as `network.into()`:
https://github.com/ZcashFoundation/zebra/blob/6f3be8fc70531849b9f02c0d9808eb9d850872e2/zebra-chain/src/primitives/address.rs#L53-L60

Please remove this duplicate consensus-critical code.

Reply:
Sorry, i missed this last night. fixed at https://github.com/ZcashFoundation/zebra/pull/6171/commits/6a68a98521f38a9a33ef89ec8ea04e02c35dd10b